seismicpro/src/refractor_velocity/refractor_velocity.py

```diff
@@ -1,539 +1,700 @@
+                   "Pass t0, x1, ..., x{N-1}, v1, ..., v{N} to define an N-layer model for N >= 2.")
         n_refractors = len(params) // 2
-        piecewise_offsets = np.zeros(n_refractors + 1)
-        piecewise_times = np.zeros(n_refractors + 1)
-        piecewise_offsets[-1] = max_offset
-        piecewise_times[0] = params[0]
-        piecewise_offsets[1:n_refractors] = params[1:n_refractors]
-        for i in range(n_refractors):
-            piecewise_times[i + 1] = ((piecewise_offsets[i + 1] - piecewise_offsets[i]) / params[n_refractors + i]) + \
-                                     piecewise_times[i]
-        return piecewise_offsets, piecewise_times
+        if n_refractors < 1:
+            raise ValueError(err_msg)
+        wrong_keys = set(get_param_names(n_refractors)) ^ params.keys()
+        if wrong_keys:
+            raise ValueError(err_msg)
 
-    def loss_piecewise_linear(self, args, loss='L1', huber_coef=20):
-        """Calculate the result of the loss function based on the passed args.
+    @classmethod
+    def _validate_params(cls, params, max_offset=None, min_velocity_step=0, min_refractor_size=0):
+        """Check if `params` dict describes a valid velocity model."""
+        cls._validate_params_names(params)
+        n_refractors = len(params) // 2
+        param_values = np.array([params[name] for name in get_param_names(n_refractors)])
+        if max_offset is None:
+            max_offset = np.inf
 
-        Method calls `calc_knots_by_params` to calculate piecewise linear attributes of a RefractorVelocity instance.
-        After that, the method calculates the loss function between the true first breaks times stored in the
-        `self.fb_times` and predicted piecewise linear function. The loss function is calculated at the offsets points.
+        negative_params = {key: val for key, val in params.items() if val < 0}
+        if negative_params:
+            raise ValueError(f"The following parameters contain negative values: {negative_params}")
 
-        Piecewise linear function is defined by the given `args`. `args` should be list-like and have the following
-        structure:
-            args[0] : intercept time in milliseconds.
-            args[1:n_refractors] : cross offsets points in meters.
-            args[n_refractors:] : velocities of each layer in kilometers/seconds.
-            Total length of args should be n_refractors * 2.
+        if np.any(np.diff(param_values[1:n_refractors], prepend=0, append=max_offset) < min_refractor_size):
+            raise ValueError(f"Offset range covered by refractors must be no less than {min_refractor_size} meters")
 
-        Notes:
-            * 'init', 'bounds' and 'params' store velocity in m/s unlike args for `loss_piecewise_linear`.
-            * The list-like `args` is due to the `scipy.optimize.minimize`.
+        if np.any(np.diff(param_values[n_refractors:]) < min_velocity_step):
```

Comment:
```suggestion
        if np.any((np.diff(param_values[n_refractors:]) < min_velocity_step) *
            ~np.isclose(np.diff(param_values[n_refractors:]) - min_velocity_step, 0)):
```
To avoid the raising ValueError if the scipy precision is not.